### PR TITLE
Various improvements to logging inside of jobs

### PR DESCRIPTION
**Pull request by ferristseng**
Related to #15649 

Merge after #15658 

### Description

  * Turns off colored logging in production (removes ANSI color codes from logging statements)
  * BGS, Rails, Shoryuken, and `ActiveRecord::Base` all log with the same format inside of jobs
    * Job logs and app server logs can now be fully parsed in [CloudWatch Log Insights](https://console.amazonaws-us-gov.com/cloudwatch/home?region=us-gov-west-1#logsV2:logs-insights$3FqueryDetail$3D$257E$2528end$257E$25272020-12-04T04*3a59*3a59.000Z$257Estart$257E$25272020-12-03T05*3a00*3a00.000Z$257EtimeType$257E$2527ABSOLUTE$257Etz$257E$2527Local$257EeditorString$257E$2527fields*20*40timestamp*2c*20*40message*0a*7c*20parse*20*22*2a*20*2a*20*2a*20ActiveJob*2f*2a*2f*2a*2f*2a*20*2a*3a*20*2a*22*20as*20timestamp*2c*20pid*2c*20thread_id*2c*20job_class*2c*20queue_name*2c*20msg_id*2c*20level*2c*20message*0a*7c*20sort*20*40timestamp*20asc*0a*7c*20display*20level*2c*20message$257EisLiveTail$257Efalse$257EqueryId$257E$25274c6942d6-b218-4958-95b1-97724144fdc2$257Esource$257E$2528$257E$2527dsva-appeals-certification-uat*2fopt*2fcaseflow-certification*2fsrc*2flog*2fcaseflow-certification-sqs-worker.out$2529$2529)
  * Moves Sentry tagging to a middleware class, and associates a AWS log url with each Sentry event

### Testing Plan

- [x] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

#### Sentry - Old

Screen Shot 2020-12-03 at 10 59 57 AM

##### Sentry - New

Screen Shot 2020-12-03 at 10 38 27 AM

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ gem "activerecord-import"
 gem "acts_as_tree"
 # BGS
 
-gem "bgs", git: "https://github.com/department-of-veterans-affairs/ruby-bgs.git", ref: "128938e80b083588164233b1db91ea7b677a2627"
+gem "bgs", git: "https://github.com/department-of-veterans-affairs/ruby-bgs.git", ref: "76ef58c10fff0bb496999aeae40f0af58de46bb6"
```
Comment (ferristseng) [inline Gemfile:12]: Integrates this change: https://github.com/department-of-veterans-affairs/ruby-bgs/pull/109

File: app/jobs/application_job.rb [hunk 2]
```diff
@@ -45,9 +45,6 @@ def capture_exception(error:, extra: {})
   # :nocov:
 
   before_perform do |job|
```
Comment (tomas-nava) [inline app/jobs/application_job.rb:47]:
```suggestion
  before_perform do
```
not using `job` anymore

File: app/jobs/middleware/job_sentry_scope_middleware.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class JobSentryScopeMiddleware
+  # :nocov:
+  def call(_worker, queue, sqs_msg, body)
+    # The structure for the `body` is described here:
+    #
+    #   https://github.com/phstc/shoryuken/wiki/Sending-a-message
+    job_class = body["job_class"].constantize
+    job_id = body["job_id"]
```
Comment (tomas-nava) [inline app/jobs/middleware/job_sentry_scope_middleware.rb:10]: `job_id` isn't being referenced anywhere

**Review by tomas-nava — APPROVED** [review]
approve!

I suggested some changes that probably would've been caught by codeclimate when you un-stacked the PR.

**Pull request closed without merging**